src/server/internalServer.cpp

```diff
@@ -772,7 +774,14 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
 
   const std::shared_ptr<Reader> reader = get_reader(bookName);
   if (reader == nullptr) {
-    return Response::build_404(*this, request, bookName);
+    MustacheData data;
+    data.set("root", m_root);
+    data.set("pattern", pattern);
+    std::string searchURL = m_root+"/search?pattern="+pattern;
+    kainjow::mustache::mustache tmpl("Make a full text search for <a href=\""+searchURL+"\">"+kiwix::urlDecode(pattern)+"</a>");
+    std::string details = tmpl.render(data);
```

Comment:
Such usage of mustache doesn't make sense - is `details` different from the contents of `tmpl`¹? `tmpl` must stay a meaningful mustache template with two parameters - `url` and `pattern` (the former must be expanded as is, while the latter must be HTML-encoded).

¹ Actually it can be, if either `searchURL` or the URL decoded pattern contain `{{m_root}}` or `{{pattern}}`, but I bet this is not why you are using mustache here.